### PR TITLE
fix(roagen): use git from pkgs

### PR DESCRIPTION
**Pull request by Moraxyc**
`git` should be taken from pkgs

### CODE REVIEW

**Review by MarcelCoding — APPROVED** [review]

**Comment by Moraxyc** [comment]
by the way, `https://git.dn42.dev/dn42/registry.git` is not publicly accessible, how to deal with it with this module?

**Comment by SuperSandro2000** [comment]
You need to create an account and then you can clone the repo. I don't think accounts get approved, so we can just use ssh cloning.

File: modules/roagen.nix [hunk 1]
```diff
@@ -33,11 +33,11 @@ in
 
         cd /tmp
         if [ -e registry ]; then
-          git clone --depth=1 https://git.dn42.dev/dn42/registry.git
+          ${lib.getExe pkgs.git} clone --depth=1 https://git.dn42.dev/dn42/registry.git
```
Comment (SuperSandro2000) [inline modules/roagen.nix:36]: Not sure if we should use gitMinimal instead

**Comment by Moraxyc** [comment]
@SuperSandro2000 yeah I know. I mean when use this module and deploy remotely， we may need to provide an option to clone the repo with an access token. Not all hosts have ssh keys with access rights.

**Comment by SuperSandro2000** [comment]
Ah, now I get what you mean. I initially thought how do we manage it via a flake input.
Which might not be fitting for this repo but could be a recommended strategy to deploy?

Otherwise maybe we need to move the systemd service to a separate user and then people can configure gitconfig/ssh_config for that user to allow cloning the repo?

**Comment by Moraxyc** [comment]
> I initially thought how do we manage it via a flake input.

I'm not sure if I understand what you mean. I do manage this module through a flake input.

We can choose ssh or https protocols, corresponding to ssh key authentication and token authentication respectively to clone a repo.

From a simple perspective, I suggest that we can add a `tokenFike` option to the roagen module, so that we can import this access token through agenix or sops-nix. 
```
git clone --depth=1 https://''$(cat ${cfg.tokenFile})@git.dn42.dev/dn42/registry.git
```

If we choose ssh to verify the identity, we may need to provide an option like `sshKeyFile`, and then configure the private key of user `bird2`.

**Comment by SuperSandro2000** [comment]
> From a simple perspective, I suggest that we can add a `tokenFike` option to the roagen module, so that we can import this access token through agenix or sops-nix.
> 
> ```
> git clone --depth=1 https://''$(cat ${cfg.tokenFile})@git.dn42.dev/dn42/registry.git
> ```

That would expose the token in the cmd arguments which usually should be avoided. I think writing a short sshconfig and writing the path to the key in there is the way to go.

**Comment by Moraxyc** [comment]
In my impression, this seems to be a common practice in nixpkgs. After the string is written into the shell script, it should be as follows.
```shell
git clone --depth=1 https://$(cat /run/agenix/dn42-git)@git.dn42.dev/dn42/registry.git
```
This would not expose the token.

However, I also agree that it could also be an option to provide such a `gitconfig` for the user bird2 to use to clone the registry.

**Comment by SuperSandro2000** [comment]
#12

**Pull request merged**